File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/maximum-number-of-integers-to-choose-from-a-range-i.cpp
class Solution {
public:
    int maxCount(vector<int>& banned, int n, int maxSum) {
        ios::sync_with_stdio(false);
        cin.tie(nullptr);

        map<int,int> mpps;
        for(auto i:banned){
            mpps[i]++;
        }
        int sum=0;
        int count=0;
        for(int i=1;i<=n;i++){
            if(mpps.find(i)==mpps.end()  && sum+i <=maxSum){
                sum+=i;
                count++;
            }

            if(sum>maxSum){
                return count;
            }
        }
        return count;
    }
};

File: 2640-maximum-number-of-integers-to-choose-from-a-range-i/maximum-number-of-integers-to-choose-from-a-range-i.java
class Solution {
    public int maxCount(int[] banned, int n, int maxSum) {

        HashMap<Integer, Boolean> bannedMap = new HashMap<>();
        for (int num : banned) {
            bannedMap.put(num, true);
        }
        int sum = 0;
        int count = 0;
        for (int i = 1; i <= n; i++) {
            if (!bannedMap.containsKey(i) && sum + i <= maxSum) {
                sum += i;
                count++;
            }
        }
        return count;
    }
}